zkvm/src/types.rs

```diff
@@ -228,6 +249,15 @@ impl Data {
         }
     }
 
+    /// Downcast the data item to a `ProgramWitness` type.
+    pub fn to_program_witness(self) -> Result<ProgramWitness, VMError> {
+        match self {
+            Data::Opaque(data) => Ok(ProgramWitness::Opaque(data)),
+            Data::ProgramWitness(pw) => Ok(pw),
```

Comment:
per comment above (keeping ::Opaque and ::Program only, w/o ::ProgramWitness), this becomes 
```suggestion
            Data::Program(p) => Ok(ProgramWitness::Program(p)),
```